ApiDocs.Validation/Params/RequestDefinitionExtensions.cs

```diff
@@ -137,6 +141,73 @@ public static void RewriteRequestWithParameters(this HttpRequest request, IEnume
             }
         }
 
+        /// <summary>
+        /// Apply any namespace translations defined in the account to the parameters in the request body
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="account"></param>
+        public static void RewriteRequestBodyNamespaces(this HttpRequest request, IServiceAccount account)
+        {
+            if (account.Transformations?.Request?.Properties == null)
+                return;
+
+            if (request.IsMatchingContentType("application/json"))
+            {
+                var translatedBody = JsonRewriter.RewriteJsonProperties(request.Body, account.Transformations.Request.Properties);
+                request.Body = translatedBody;
+            }
+            else if (request.IsMatchingContentType("multipart/related"))
+            {
+                var parsedBody = new MultipartMime.MultipartMimeContent(request.ContentType, request.Body);
+                var partsToRewrite = parsedBody.PartsWithContentType("application/json");
+                foreach(var part in partsToRewrite)
+                {
+                    part.Body = JsonRewriter.RewriteJsonProperties(part.Body, account.Transformations.Request.Properties);
+                }
+                request.Body = parsedBody.ToString();
+            }
+        }
+
+        public static void RewriteResponseBodyNamespaces(this HttpResponse response, IServiceAccount account)
+        {
+            if (account.Transformations?.Response?.Properties == null)
+                return;
+            if (!response.IsMatchingContentType("application/json"))
+                return;
+            if (response.Body.Length == 0)
+                return;
+
+            var translatedBody = JsonRewriter.RewriteJsonProperties(response.Body, account.Transformations.Response.Properties);
+            response.Body = translatedBody;
+        }
+
+        public static Dictionary<string, string> RemoveKnownEntities(this Dictionary<string, string> namespaceMap)
+        {
+            var knownKeys = new string[] { "!action" };
+            var output = new Dictionary<string, string>(namespaceMap);
+            foreach(var key in knownKeys)
+            {
+                output.Remove(key);
+            }
+            return output;
+        }
+
+        /// <summary>
+        /// Swap the keys and values in a dictionary (values become keys, keys become values).
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="dictionary"></param>
+        /// <returns></returns>
+        private static Dictionary<T, T> SwapKeysAndValues<T>(this Dictionary<T, T> dictionary)
+        {
+            var output = new Dictionary<T, T>();
```

Comment:
what a wasted perfect opportunity for an unreadable linq one-liner!
`return dictionary.ToDictionary(kvp=>kvp.Value,kvp=>kvp.Key);`